nni/algorithms/compression/v2/pytorch/utils/evaluator.py

```diff
@@ -0,0 +1,713 @@
+
+    def register_hooks(self, hooks: List[Hook]):
+        """
+        NOTE: NNI users are not recommended to call this function, it's used by NNI compressor.
+
+        The input is a list of ``TensorHook``, ``ForwardHook``, ``BackwardHook``,
+        please view how to use ``TensorHook``, ``ForwardHook``, ``BackwardHook``.
+        This function will call ``Hook.register()`` of hook in ``hooks``, and record the hook in ``self._hooks``.
+        """
+        if not hasattr(self, '_hooks'):
+            self._hooks: List[Hook] = []
+        for hook in hooks:
+            hook.register()
+            self._hooks.append(hook)
+
+    def get_all_hooks(self) -> List[Hook]:
+        """
+        NOTE: NNI users are not recommended to call this function, it's used by NNI compressor.
+
+        Get all registered ``Hook``.
+        """
+        return getattr(self, '_hooks', [])
+
+    def remove_all_hooks(self):
+        """
+        NOTE: NNI users are not recommended to call this function, it's used by NNI compressor.
+
+        Call ``Hook.remove()`` of all ``Hook`` instances in ``self._hooks``, then clear ``self._hooks``.
+        """
+        if hasattr(self, '_hooks'):
+            for hook in self._hooks:
+                hook.remove()
+            self._hooks.clear()
+
+    def train(self, max_steps: int | None = None, max_epochs: int | None = None):
+        """
+        NOTE: NNI users can call this function to train the bound model, but not necessary.
+
+        Train the bound model with default optimization loop defined by user and only change the training duration.
+        """
+        raise NotImplementedError
+
+    def finetune(self):
+        """
+        NOTE: NNI users can call this function to finetune the bound model, but not necessary.
+
+        Finetune the bound model with default optimization loop defined by user.
+        """
+        raise NotImplementedError
+
+    def evaluate(self) -> float | Tuple[float, Any]:
+        """
+        NOTE: NNI users can call this function to evaluate the bound model, but not necessary.
+
+        Evaluate the bound model, the returned metric should be a float number,
+        or a tuple of two elements with the first element being a float number.
+        NNI will take this float number as the metric used in the compression algorithm.
+        """
+        # Note that the first item of the returned value will be used as the default metric used by NNI.
+        raise NotImplementedError
+
+    def get_dummy_input(self) -> Any:
+        """
+        NOTE: NNI users are not recommended to call this function, it's used by NNI compressor.
+
+        The returned value is a dummy input for the model, always used by ``torch.jit.trace``.
+        """
+        raise NotImplementedError
+
+
+class LightningEvaluator(Evaluator):
+    """
+    NOTE: If the the test metric is needed by nni, please make sure log metric with key ``default`` in LightningModule.test_step().
```

Comment:
would be better to add short description for this class, then the NOTE message follows the description. for example, the description could be "LightningEvaluator is the Evaluator based on PytorchLightning. It is very friendly to the users who are familiar to PytorchLightning or already have training/validation/testing code written in PytorchLightning."

Reply:
good suggestion, will add them.